cloud-build/presubmit.sh

```diff
@@ -16,47 +16,45 @@ echo "Changed files: ${CHANGED_FILES[*]}"
 
 # Determines whether a given string is a prefix string of any changed file name
 is_prefix() {
-  for file in "${CHANGED_FILES[@]}"
-  do
-    if [[ $file =~ ^$1 ]]
-    then
+  for file in "${CHANGED_FILES[@]}"; do
+    if [[ $file =~ ^$1 ]]; then
       return 0
     fi
   done
   return 1
 }
 
 # Determines init actions directories that were modified
+RUN_ALL_TESTS=false
 declare -a DIRECTORIES_TO_TEST
-for dir in */
-do
-  # Skip not init action changes
-  if [[ $dir =~ ^(integration_tests/|util/|cloud-build/)$ ]]
-  then
-    continue
+for dir in */; do
+  # Run all tests if common directories were modified
+  if [[ $dir =~ ^(integration_tests/|util/|cloud-build/)$ ]]; then
+    echo "All tests will be run: '$dir' was modified"
+    RUN_ALL_TESTS=true
+    break
   fi
-  if is_prefix "$dir"
-  then
+  if is_prefix "$dir"; then
     DIRECTORIES_TO_TEST+=("$dir")
   fi
 done
 echo "Test directories: ${DIRECTORIES_TO_TEST[*]}"
 
 # Determines what tests in modified init action directories to run
-declare -a ALL_TESTS
-for test_dir in "${DIRECTORIES_TO_TEST[@]}"
-do
-  if ! tests=$(compgen -G "${test_dir}test*.py")
-  then
+declare -a TESTS_TO_RUN
+for test_dir in "${DIRECTORIES_TO_TEST[@]}"; do
+  if ! tests=$(compgen -G "${test_dir}test*.py"); then
     echo "ERROR: presubmit failed - can not find tests inside '${test_dir}' directory"
     exit 1
   fi
   mapfile -t tests_array < <(echo "${tests}")
-  ALL_TESTS+=("${tests_array[@]}")
+  TESTS_TO_RUN+=("${tests_array[@]}")
 done
-echo "Tests: ${ALL_TESTS[*]}"
+echo "Tests: ${TESTS_TO_RUN[*]}"
 
 # Run tests of the init actions that were modified
-python -m unittest "${ALL_TESTS[@]}"
-
-exit $?
+if [[ $RUN_ALL_TESTS == true ]]; then
+  python -m unittest
```

Comment:
Are you sure this runs all tests? Running this command in my machine results in no tests being run.

Reply:
Hm, I tested with `fastunit` and it runs all tests from the git repo root. Maybe there are difference between `unittest` and `fastunit`.

Anyway, we need to use `fastunit` because without it it will take hours to run all tests.

May you configure to use `fastunit` here?

Reply:
Yes, I will work on it.